### PR TITLE
Image Popup in MainImageModule IMPROVED:

### PR DESCRIPTION
**Pull request by NitinBhasneria**
Fixes : #431

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `grunt jasmine`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] PR body includes `fixes #0000`-style reference to original issue #
* [x] ask `@publiclab/reviewers` for help, in a comment below

Screenshot from 2020-03-28 17-21-40

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by NitinBhasneria** [comment]
@VladimirMikulic @cesswairimu @jywarren My previous two PR's having the same problem in `grunt jasmine`. Please have a look.

**Comment by VladimirMikulic** [comment]
@NitinBhasneria I've checked it out know. The failing tests are not your fault. `mkdirp` package uses object spread operator (`...`) which is not supported on Node < 8.3.
We run Travis on Node 5, 6, 7 and it fails.

**Comment by VladimirMikulic** [comment]
Please post a gif of your change.

**Comment by NitinBhasneria** [comment]
@VladimirMikulic Here while uploading image I encountered another problem.
I am making an issue for that please have a look #436

**Comment by NitinBhasneria** [comment]
@VladimirMikulic while changing the things in the src folder in editor's code. The changes are reflected in example/index.html but it is not reflected in http://localhost:3000/post , even after `grunt` and `yarn install --force`. I also change path of editor in plots2/packagae.json

**Comment by VladimirMikulic** [comment]
You can simply copy the PublicLab Editor to the node_modules of `plots2` and that's it.

**Comment by NitinBhasneria** [comment]
I have copied the PublicLab.Editor in `plots2/public/lib` but still the changes are not reflected in `http://localhost:3000/post`

**Comment by NitinBhasneria** [comment]
In plots2/public/lib there is also a folder named publiclab-editor. Is there something to do with this too. I am really confused with these packages. Can you explain what to do?

**Comment by VladimirMikulic** [comment]
Correct, it's `lib`, not `node_moduless`, sorry.

**Comment by NitinBhasneria** [comment]
Bur still the changes are not reflecting. Can you help? @VladimirMikulic

**Comment by VladimirMikulic** [comment]
I don't have `plots2` locally, but it if it doesn't work, it means that your code fails, which is most likely the case, since you assign a value in an `if` statement...

**Comment by NitinBhasneria** [comment]
Actually I even tried to give output in console but no output is shown.🙄

**Comment by VladimirMikulic** [comment]
Are you sure that you've compiled your changes and copied the old `dist` file with the updated one?

**Comment by NitinBhasneria** [comment]
Yes, I m sure. I have tried many times.

**Comment by VladimirMikulic** [comment]
I think I know why. Your JS is inside of HTML file which is not served on the client-side in `plots2`. Your change needs to be in `.js` file.

**Comment by NitinBhasneria** [comment]
Ok I will update you after further tests thanks

**Comment by NitinBhasneria** [comment]
Also I want to ask do I have to do yarn install after tests.

**Comment by VladimirMikulic** [comment]
You don't need `yarn install` since you manually copy files.

**Comment by NitinBhasneria** [comment]
Also there is pre existing publiclab-editor folder do er have to replace that?

**Comment by VladimirMikulic** [comment]
My zsh history says that the file that needs to be replaced is `plots2/public/lib/publiclab-editor/dist/PublicLab.Editor.js`

**Comment by NitinBhasneria** [comment]
Ok thanks, I will test and then update you surely

**Comment by NitinBhasneria** [comment]
@VladimirMikulic Thanks it worked.

**Comment by VladimirMikulic** [comment]
That's it @NitinBhasneria! Wohoo.

**Comment by NitinBhasneria** [comment]
@VladimirMikulic please have a look and approve this pr.
editor3

File: src/modules/PublicLab.MainImageModule.js [hunk 1]
```diff
@@ -127,6 +128,7 @@ module.exports = PublicLab.MainImageModule = PublicLab.Module.extend({
       progressall: function (e, data) {
 
         var progress = parseInt(data.loaded / data.total * 100, 10);
+        dragImageI.innerHTML = "";
```
Comment (VladimirMikulic) [inline src/modules/PublicLab.MainImageModule.js]: Could you please add a comment here, so x months later down the road we know why we did it? Thank you.

**Comment by NitinBhasneria** [comment]
@jywarren ready for your review

**Comment by keshav234156** [comment]
@NitinBhasneria I have addressed this  issue #396 .I think we can close this issue

**Comment by jywarren** [comment]
Hi, just to disambiguate -- thanks @keshav234156 -- does this PR hide the "drag to upload" text whereas @keshav234156's PR preserves the image aspect ratio? So perhaps both are needed? Thanks a lot, if you can both help clarify exactly what your PR does, we can probably make use of both! Thanks!!! 🎉

**Comment by keshav234156** [comment]
@jywarren pr #437 fixes both of the issuesaspeck ratio as well as hiding of drag to drop text

**Comment by keshav234156** [comment]
This pr fixes the hiding of drag to upload text while uploading

**Comment by jywarren** [comment]
Ah, i'm sorry, i merged the other one first. Apologies for missing the parallel work here, I appreciate the contributions from everyone! @NitinBhasneria if you'd like to resolve conflicts and re-confirm that this works, I can merge it so everybody gets credited. Just be sure that the redundant parts of the code don't interfere destructively! Thank you @NitinBhasneria and @keshav234156 and @VladimirMikulic and @govindgoel!

**Comment by NitinBhasneria** [comment]
@jywarren done with conflicts. Thanks

File: src/modules/PublicLab.MainImageModule.js [hunk 2]
```diff
@@ -132,6 +133,11 @@ module.exports = PublicLab.MainImageModule = PublicLab.Module.extend({
...
+        
+        // For hiding the HTML "Drag an image here to upload." after uploading image.
+        dragImageI.innerHTML = "";
```
Comment (jywarren) [inline src/modules/PublicLab.MainImageModule.js:139]: Oh hm, is this line duplicated form 136? And, would you mind uploading a GIF of this working just so we can be sure? Thank you!

**Comment by NitinBhasneria** [comment]
@jywarren Duplicate removed. here is the gif. 
Peek 2020-04-22 16-20

**Comment by jywarren** [comment]
Hooray! Thanks a huge amount. this is super great and thank you for sticking with it through a very careful and detailed review and revisions. Merging now!

**Pull request merged**